server/service/frontend.go

```diff
@@ -70,14 +71,24 @@ func ServeFrontend(urlPrefix string, sandbox bool, logger log.Logger) http.Handl
 	})
 }
 
-func ServeEndUserEnrollOTA(urlPrefix string, logger log.Logger) http.Handler {
+func ServeEndUserEnrollOTA(svc fleet.Service, urlPrefix string, logger log.Logger) http.Handler {
 	herr := func(w http.ResponseWriter, err string) {
 		logger.Log("err", err)
 		http.Error(w, err, http.StatusInternalServerError)
 	}
 
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		writeBrowserSecurityHeaders(w)
+		setupRequired, err := svc.SetupRequired(r.Context())
+		if err != nil {
+			herr(w, "setup required err: "+err.Error())
+			return
+		}
+
+		if setupRequired {
+			herr(w, "fleet instance not setup")
```

Comment:
Nice! This fixes that panic. We should probably get some product input on showing a nicer UI here (the browser just shows this string), but I think that can be addressed separately.